scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -191,16 +192,15 @@ object ContextAndArgs {
   def withParser[T](parser: ArgsParser[Try]): Array[String] => (ScioContext, T) =
     args =>
       parser.parse(args) match {
-        // scalastyle:off regex
         case Failure(exception) =>
-          Console.err.println(exception.getMessage)
-          sys.exit(1)
+          throw exception
         case Success(Left(usageOrHelp)) =>
+          // scalastyle:off regex
           Console.out.println(usageOrHelp)
-          sys.exit(0)
+          // scalastyle:on regex
+          throw new UsageOrHelpException()
```

Comment:
We could attach `UncaughtExceptionHandler` to current thread here to skip printing `UsageOrHelpException` line from the output.
Do we want to?